build/DownloadCFamilyPlugin/DownloadAndExtract.cs

```diff
@@ -84,188 +79,43 @@ public override bool Execute()
         {
             LogMessage($"Download url: {DownloadUrl}");
 
-            var pluginFileName = ExtractPluginFileName(DownloadUrl);
+            var pluginFileName = Common.ExtractPluginFileNameFromUrl(DownloadUrl, Log);
 
             // Ensure working directories exists
             var localWorkingFolder = Path.Combine(Environment.GetEnvironmentVariable("LocalAppData"), "SLVS_CFamily_Build");
             var perVersionPluginFolder = Path.Combine(localWorkingFolder, Path.GetFileNameWithoutExtension(pluginFileName));
-            EnsureWorkingDirectoryExist(perVersionPluginFolder);
+            Common.EnsureWorkingDirectoryExist(perVersionPluginFolder, this.Log);
 
             // Download and unzip the jar
             var jarFilePath = Path.Combine(perVersionPluginFolder, pluginFileName);
-            DownloadJarFile(DownloadUrl, jarFilePath);
-            UnzipJar(jarFilePath, perVersionPluginFolder);
+            Common.DownloadJarFile(DownloadUrl, jarFilePath, Log);
+            Common.UnzipJar(jarFilePath, perVersionPluginFolder, Log);
 
             // Uncompress and extract the windows tar archive to get the subprocess exe
-            var tarFilePath = FindSingleFile(perVersionPluginFolder, WindowsTxzFilePattern);
+            var tarFilePath = Common.FindSingleFile(perVersionPluginFolder, WindowsTxzFilePattern, Log);
             var tarSubFolder = Path.Combine(perVersionPluginFolder, TarUnzipSubFolder);
-            UncompressAndUnzipTar(tarFilePath, tarSubFolder);
+            Common.UncompressAndUnzipTgx(tarFilePath, tarSubFolder, Log);
 
             // Locate the required files from the uncompressed jar and tar
             var fileList = FindFiles(perVersionPluginFolder);
 
             FilesToEmbed = fileList.ToArray();
-
             return !Log.HasLoggedErrors;
         }
 
-        private string ExtractPluginFileName(string url)
-        {
-            if (!url.EndsWith(".jar", StringComparison.InvariantCultureIgnoreCase))
-            {
-                throw new ArgumentException("Expecting the url to end with '.jar'");
-            }
-
-            var fileName = url.Split('/').Last();
-
-            LogMessage($"Plugin file name: {fileName}");
-            return fileName;
-        }
-
         private List<string> FindFiles(string searchRoot)
         {
             var files = new List<string>();
 
-            foreach(var file in SingleFilePatterns)
-            {
-                files.Add(FindSingleFile(searchRoot, file));
-            }
+            files.AddRange(Common.FindSingleFiles(searchRoot, SingleFilePatterns, Log));
+            files.AddRange(Common.FindMultipleFiles(searchRoot, MultipleFilesPatterns, Log));
```

Comment:
I tested that the results of the old and new methods are the same - I ran a version of the task with both old and new versions of the code with programmatic checks that exactly the same files are returned; then I deleted the old version of the code.